### ISSUE TITLE
toFileStream silent failure with too large string

### ISSUE THREAD

**Issue by skibz** [soldair/node-qrcode#83]
```javascript
// big_string is about 2 ^ 16 - 1 characters in length
var big_string = get_big_string_somehow()
var stream = get_writable_stream_somehow()
qr.toFileStream(stream, big_string, console.log)
```

when this is program is run, `console.log` is never invoked. what i expected is `console.log` to be invoked with an error describing that there is too much input data to be encoded.

note that i'm not sure **exactly** how big the string has to be to cause the silent failure, but the project in which i am using this library was attempting to use a string of more or less that size.

**Comment by vigreco** [comment]
Hi @skibz,
`toFileStream` hasn't a callback parameter, indeed there's an error in documentation. My mistake.

You should use stream's events instead.

The correct signature is 
```javascript
function toFileStream (stream, text, opts)
```
I'll fix it ASAP.

Thanks for catching that!

**Comment by skibz** [comment]
ah, now i feel silly for not reading the source code myself 😛 
thanks for clearing this up for me @vigreco

**Comment by soldair** [comment]
another note. I think that your very large string wont fit in a qrcode of any size.
https://github.com/soldair/node-qrcode#qr-code-capacity

i dont know really recall what happens if you do this though. =)

**Comment by skibz** [comment]
okay, so then i should be checking the limits of the input data using the table @soldair referenced, and the streams method has no callback.

thanks guys 👍

**Comment by vigreco** [comment]
@skibz, the lib already checks for you if the text can fit in a qr code.
If you don't specify a qr code version, the more suitable will be used.
If your text size exceeds the max allowed size an error will be throw.

That said, I checked the code and I found a couple of issues that prevent the error to be propagated when using the `toFileStream` api.
So if you are using the `error` event of WritableStream to check for errors, you will see nothing related to qrcode.
Sorry for that, I'll fix this by tomorrow and let you know.
